### ISSUE TITLE
lr schedulers step function should not have any arguments

### ISSUE THREAD

**Issue by gkouros** [KevinMusgrave/pytorch-metric-learning#358]
Stepping lr schedulers should have no arguments. Old pytorch api had an argument epoch, not curr_accuracy as in  function [save_models_and_eval](https://github.com/KevinMusgrave/pytorch-metric-learning/blob/91704181f0f150cb3bff7fba36f4bb0fbf87402a/src/pytorch_metric_learning/utils/logging_presets.py#L154). 
https://github.com/KevinMusgrave/pytorch-metric-learning/blob/91704181f0f150cb3bff7fba36f4bb0fbf87402a/src/pytorch_metric_learning/utils/logging_presets.py#L183

PyTorch 1.9,  in particular, gives a deprecation warning. See [_LRScheduler](https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html#_LRScheduler). 
```
.../lib/python3.8/site-packages/torch/optim/lr_scheduler.py:154: UserWarning: The epoch parameter in `scheduler.step()` 
was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the 
deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. 
Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
```

**Comment by KevinMusgrave** [comment]
But ```ReduceLROnPlateau``` does take an argument: https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html

The regular lr schedulers are stepped here without any arguments:
https://github.com/KevinMusgrave/pytorch-metric-learning/blob/91704181f0f150cb3bff7fba36f4bb0fbf87402a/src/pytorch_metric_learning/trainers/base_trainer.py#L174-L184

**Comment by gkouros** [comment]
Indeed, you are right. I replaced the ReduceLROnPlateau schedulers with the StepLR schedulers and forgot to update the keys in the dictionary from _plateau to _epoch, and that's why I had the warning. Thanks for the assistance. I'm closing the (non)issue.
